torch/csrc/jit/tensorexpr/external_functions.cpp

```diff
@@ -1268,25 +1268,67 @@ void nnc_aten_adaptive_avg_pool2d(
     int64_t* buf_dims,
     int64_t* buf_strides,
     int8_t* buf_dtypes,
-    int64_t args_num,
+    int64_t,
     int64_t* extra_args) {
+  const int64_t x_qdtype = extra_args[2];
+  c10::optional<std::vector<std::pair<size_t, QIData>>> qdata;
+  if (x_qdtype != -1) {
+    qdata = {{1u,
+              {((double*)extra_args)[0],
+                extra_args[1],
+                at::toQIntType(static_cast<c10::ScalarType>(x_qdtype))}}};
+  }
   auto tensors = constructTensors(
-      bufs_num, buf_data, buf_ranks, buf_dims, buf_strides, buf_dtypes);
+      bufs_num, buf_data, buf_ranks, buf_dims, buf_strides, buf_dtypes, qdata);
 
   at::Tensor& r = tensors[0];
-  const at::Tensor& x = tensors[1];
-  int64_t H = extra_args[0];
-  int64_t W = H;
-  if (args_num > 1) {
-    W = extra_args[1];
-  }
   try {
-    r = at::adaptive_avg_pool2d(x, {H, W});
+    r = at::adaptive_avg_pool2d(tensors[1], {extra_args[3], extra_args[4]});
```

Comment:
To make the code more readable, how about mark the meaning of `extra_args[3]` and `extra_args[4]`? Something like, `/* H= */ extra_args[3]`?

Reply:
Good suggestion, I will update the line.